tests/unit/profiles-management/helpers/test_k8s.py

```diff
@@ -0,0 +1,31 @@
+from lightkube.generic_resource import GenericNamespacedResource
+from lightkube.models.meta_v1 import ObjectMeta
+
+from profiles_management.helpers.k8s import get_annotations, to_rfc1123_compliant
+
+
+def test_annotations():
+    resource = GenericNamespacedResource(
+        metadata=ObjectMeta(name="test", annotations={"test": "value"})
+    )
+
+    assert get_annotations(resource) == {"test": "value"}
+
+
+def test_no_annotations():
+    resource = GenericNamespacedResource(metadata=ObjectMeta(name="test"))
+
+    assert get_annotations(resource) == {}
+
+
+def test_non_alphanum_to_hyphen():
+    name = "kimonas@canonical.com"
+
+    assert to_rfc1123_compliant(name) == "kimonas-canonical-com"
+
+
+def test_more_than_63_char_name():
+    name = "I-had-to-think-of-a-reeeally-long-string-to-use-for-the-test-which-was-tough"
+
+    assert len(name) > 63
+    assert len(to_rfc1123_compliant(name)) == 63
```

Comment:
These two can be parametrised. You could also add other cases, such as:
* empty string ("")
* all alphas ("abcdefg")
* all nums ("1234")
* all invalid chars ("$%%@#")

Reply:
great suggestion!
55527d8